packages/aws-cdk/lib/cli/cdk-toolkit.ts

```diff
@@ -1218,30 +1211,23 @@ export class CdkToolkit {
   }
 
   public async refactor(options: RefactorOptions): Promise<number> {
-    if (!options.dryRun) {
-      info('Refactor is not available yet. Too see the proposed changes, use the --dry-run flag.');
+    const tk = new Toolkit();
+
+    const casmSource = await tk.fromAssemblyBuilder(() => this.assembly());
+    try {
+      await tk.refactor(casmSource, {
+        dryRun: options.dryRun,
+        skipFile: options.skipFile,
+        stacks: {
+          patterns: options.selector.patterns,
+          strategy: options.selector.patterns.length > 0 ? StackSelectionStrategy.PATTERN_MATCH : StackSelectionStrategy.ALL_STACKS,
+        },
```

Comment:
Nice! This could maybe be helper so other (future) actions can re-use it. Not important for this PR though.